sentinelhub/api/byoc.py

```diff
@@ -17,17 +17,32 @@
 from .utils import datetime_config, geometry_config, remove_undefined
 
 
+@dataclass_json(letter_case=LetterCase.CAMEL, undefined=Undefined.INCLUDE)
+@dataclass
+class ByocCollectionBand:
+    """Dataclass to hold BYOC collection band specification"""
+
+    other_data: CatchAll
+    source: Optional[str] = None
+    band_index: Optional[int] = None
+    bit_depth: int = 8
+    sample_format: str = "UINT"
+    no_data: Optional[float] = None
+
+
 @dataclass_json(letter_case=LetterCase.CAMEL, undefined=Undefined.INCLUDE)
 @dataclass
 class ByocCollectionAdditionalData:
     """Dataclass to hold BYOC collection additional data"""
 
     other_data: CatchAll
-    bands: Optional[dict] = None
+    bands: Dict[str, ByocCollectionBand] = None
```

Comment:
Because the default value is `None`, should this have a type `Optional[Dict[str, ByocCollectionBand]]`?